scripts/compareAPIs.js

```diff
@@ -2,360 +2,545 @@
   let pairs = [];
-  // we only care about changes to already published APIs, so find all matching pairs based on what's been published
+
+  // find all matching pairs based on what's been published
   for (let pubApi of publishedAPIs) {
     let pubApiPath = pubApi.split(path.sep);
     let sharedPath = path.join(...pubApiPath.slice(pubApiPath.length - 4));
-    let matchingBranchFile;
+    let found = false;
     for (let branchApi of branchAPIs) {
       if (branchApi.includes(sharedPath)) {
-        matchingBranchFile = branchApi;
+        found = true;
         pairs.push({pubApi, branchApi});
         break;
       }
     }
-    if (!matchingBranchFile) {
-      summaryMessages.push({msg: `removed module ${pubApi}`, severity: 'error'});
+    if (!found) {
+      pairs.push({pubApi, branchApi: null});
     }
   }
-  let privatePackages = [];
-  // don't care about not private APIs, but we do care if we're about to publish a new one
+
+  // don't care about private APIs, but we do care if we're about to publish a new one
   for (let branchApi of branchAPIs) {
     let branchApiPath = branchApi.split(path.sep);
     let sharedPath = path.join(...branchApiPath.slice(branchApiPath.length - 4));
-    let matchingPubFile;
+    let found = false;
     for (let pubApi of publishedAPIs) {
       if (pubApi.includes(sharedPath)) {
-        matchingPubFile = pubApi;
-        // don't re-add to pairs
+        found = true;
         break;
       }
     }
-    if (!matchingPubFile) {
-      let json = JSON.parse(fs.readFileSync(path.join(branchApi, '..', '..', 'package.json')), 'utf8');
-      if (!json.private) {
-        summaryMessages.push({msg: `added module ${branchApi}`, severity: 'warn'});
-      } else {
-        privatePackages.push(branchApi);
-      }
+    let json = JSON.parse(fs.readFileSync(path.join(branchApi, '..', '..', 'package.json')), 'utf8');
+    if (!found && !json.private) {
+      pairs.push({pubApi: null, branchApi});
     }
   }
 
-  let count = 0;
-  let diffs = {};
+  let allDiffs = {};
   for (let pair of pairs) {
-    let diff = getDiff(summaryMessages, pair);
-    if (diff.diff.length > 0) {
-      count += 1;
-      diffs[diff.name] = diff.diff;
+    let {diffs, name} = getDiff(pair);
+    if (diffs && diffs.length > 0) {
+      allDiffs[name] = diffs;
     }
   }
-  let modulesAdded = branchAPIs.length - privatePackages.length - publishedAPIs.length;
-  if (modulesAdded !== 0) {
-    summaryMessages.push({msg: `${Math.abs(modulesAdded)} modules ${modulesAdded > 0 ? 'added' : 'removed'}`, severity: modulesAdded > 0 ? 'warn' : 'error'});
-  } else {
-    summaryMessages.push({msg: 'no modules removed or added', severity: 'info'});
+
+  for (let [, diffs] of Object.entries(allDiffs)) {
+    for (let {result: diff, simplifiedName} of diffs) {
+      if (diff.length > 0) {
+        if (allChanged.has(simplifiedName)) {
+          console.log(simplifiedName, 'already in set');
```

Comment:
Why are we logging this? It seems like noise.

Reply:
i want to know if this is ever hit, it would most likely mean we have two interfaces with the same name, which would not be good for my script
it's a hunch, so I'm not failing based on it
right now, it's never hit